Ewl 3741 topic member pitch

rpama opened the pull request:
<!-- NOTE: Please just put "N/A" for any section below that isn't applicable to the work you've done, do not omit entirely. -->

## Ticket(s)

**Jira Ticket**

- [EWL-3741: Topics | SG | Design Member Pitch CTA](https://issues.ama-assn.org/browse/EWL-3741)

## Description

Refactors atom list unordered to accept json data instead of hard-coded placeholder text
Creates molecule block-cta-with-list.
Creates organism topic-rail-right containing block-cta-with-list
Adds organism topic-rail-right to template topic

## To Test

- [x] Go to Atoms > Lists > Unordered
- [x] Go to Molecule > Blocks > block-cta-with-list
- [x] Go to Organisms > Topic > topic-rail-right
- [x] Go to Templates > Topic > Topic Hub Template

Observe all the patterns look sensible and approximate the MVP from the wireframes. 
Review markup for each and see that it, too, looks sensible.


## Relevant Screenshots/GIFs
Wireframe:

<img width="266" alt="screen shot 2017-07-27 at 4 34 12 pm" src="https://user-images.githubusercontent.com/28711067/28693308-a3f43736-72e9-11e7-827e-dbb544736c51.png">


SG:
![screen shot 2017-07-27 at 4 29 37 pm](https://user-images.githubusercontent.com/28711067/28693300-94b31968-72e9-11e7-89e0-bdeec5a60e99.png)

Wireframe:

![screen shot 2017-07-27 at 4 30 50 pm](https://user-images.githubusercontent.com/28711067/28693322-afc94510-72e9-11e7-9fc8-dfa9a0ce5e3d.png)

SG:
![screen shot 2017-07-27 at 4 29 04 pm](https://user-images.githubusercontent.com/28711067/28693290-8b905350-72e9-11e7-82c2-d704bf0d1ab4.png)



## Remaining Tasks

Since we're handling things a bit more iteratively this cycle, there may be questions/imperfections that remain and that we'll catch as we build out more. Those things will be entered into new tickets.


## Additional Notes

The rail right doesn't attach to the right because it's basically waiting for other columns to come before it. I thought about floating it, but we don't really use floats much so I left it as is even though it might for the moment appear slightly confusing.

deadarm commented on styleguide/source/_patterns/00-atoms/03-lists/00-unordered/unordered.twig:
```diff
@@ -0,0 +1,14 @@
+<div class="text">
```
Pretty sure the original markup came from Pattern Lab itself, so if you want to remove the `<div>` or add a class of `list` it would make more sense.

rpama commented:
@deadarm the wireframes were inconsistent in how they presented the location of the button (whether it is within the grey block or outside of it). But you're right it looks like they got updated without warning and the version with the button outside the block simply doesn't exit anymore. So I changed the placement.

Removed that div around the `ul` too. Didn't know why it was there, but seems you're right that it's not serving any purpose.

deadarm commented on styleguide/source/_patterns/02-organisms/topic/topic-rail-right/topic-rail-right.md:
```diff
@@ -0,0 +1,6 @@
+title: "Topic Rail Right"
+---
+
+This is an aside that will site on the right-hand side of a page with a 3-column layout. It is designed exclusively for a Topic Hub page or template.
```
Typo 'site' should be 'sit'.

deadarm commented on styleguide/source/_patterns/02-organisms/topic/topic-rail-right/topic-rail-right.twig:
```diff
@@ -0,0 +1,3 @@
+<aside class="topic_rail-right">
```
Just to clarify, this organism is so that the right rail can be displayed as such in the style guide, but isn't actually used in the template?

chelsiejohnston replied:
I'm not sure it makes sense to have separate organisms for the columns/rails. The rails can have variable content. I think it would be simpler/better to include the section organisms directly in the Template pattern.

deadarm replied:
I think Rachel is adding the aside with include in the actual template too, and then also created an organism to show the rail as its own thing?

chelsiejohnston replied:
I don't think it's valuable IMO, but I suppose it couldn't hurt either. @rpama if you go this route, just make sure the `.md` explains that the pattern isn't included anywhere and is just to show what an example of a right rail looks like on its own

deadarm replied:
Ok also lmk if we should remove because I also added that concept to https://github.com/AmericanMedicalAssociation/AMA-style-guide/pull/169 after seeing Rachel's.

chelsiejohnston replied:
My vote is take 'em out but if you guys feel different then that's fine too!

chelsiejohnston replied:
Also, this LGTM, please don't let my comments prevent merging :)

deadarm commented:
It makes sense to remove it since a sidebar wouldn't be used or shown on its own. @rpama if you remove the topic-rail-right organism and fix the sit typo, I think this is can be wrapped up. I will also more the left rail organism on my PR.

deadarm commented:
Merged successfully.
